hvac/api/vault_api_category.py

```diff
@@ -36,8 +36,8 @@ def __getattr__(self, item):
         :return: The requested class instance where available.
         :rtype: hvac.api.VaultApiBase
         """
-        if item in self.implemented_class_names:
-            private_attr_name = self.get_private_attr_name(item)
+        private_attr_name = self.get_private_attr_name(item)
```

Comment:
That does not seem to do the trick. With further testing it seem the call to `self.get_private_attr_name()` also causes an infinite recursion.